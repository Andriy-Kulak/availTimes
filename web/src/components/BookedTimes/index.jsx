import React from 'react'
import PropTypes from 'prop-types'
import Header from '../Header'
import { prettifyDate } from '../../utils/dataTransform'
import TableHeader from '../Table/Header'

const BookedTimes = React.memo(({ data }) => {
  return (
    <>
      <Header text="Booked Times" />
      <table className="bookings table">
        <TableHeader data={['Advisor ID', 'Student Name', 'Date/Time']} />
        <tbody>
          {data.map((x) => (
            <tr key={`${x.id}-${x.time}-${x.name}`}>
              <td>{x.id}</td>
              <td>{x.name}</td>
              <td>
                <time dateTime={x.time}>{prettifyDate(new Date(x.time))}</time>
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </>
  )
})

BookedTimes.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string,
      time: PropTypes.string,
    })
  ),
}

export default BookedTimes
