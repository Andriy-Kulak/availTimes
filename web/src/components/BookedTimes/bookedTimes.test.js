import React from 'react'
import BookedTimes from './index'
import renderer from 'react-test-renderer'

describe('Booked Times', () => {
  it('full of data', () => {
    const data1 = [
      { id: 319369, time: '2019-08-24T20:00:00-04:00', name: '333' },
      { id: 319369, time: '2019-08-25T10:30:00-04:00', name: '333' },
    ]
    const component = renderer.create(<BookedTimes data={data1} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('with no data', () => {
    const data1 = []
    const component = renderer.create(<BookedTimes data={data1} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
