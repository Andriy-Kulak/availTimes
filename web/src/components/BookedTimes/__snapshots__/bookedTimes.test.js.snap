// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booked Times full of data 1`] = `
Array [
  <h2>
    Booked Times
  </h2>,
  <table
    className="bookings table"
  >
    <thead>
      <tr>
        <th>
          Advisor ID
        </th>
        <th>
          Student Name
        </th>
        <th>
          Date/Time
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          319369
        </td>
        <td>
          333
        </td>
        <td>
          <time
            dateTime="2019-08-24T20:00:00-04:00"
          >
            08/24/2019 8:00 PM
          </time>
        </td>
      </tr>
      <tr>
        <td>
          319369
        </td>
        <td>
          333
        </td>
        <td>
          <time
            dateTime="2019-08-25T10:30:00-04:00"
          >
            08/25/2019 10:30 AM
          </time>
        </td>
      </tr>
    </tbody>
  </table>,
]
`;

exports[`Booked Times with no data 1`] = `
Array [
  <h2>
    Booked Times
  </h2>,
  <table
    className="bookings table"
  >
    <thead>
      <tr>
        <th>
          Advisor ID
        </th>
        <th>
          Student Name
        </th>
        <th>
          Date/Time
        </th>
      </tr>
    </thead>
    <tbody />
  </table>,
]
`;
